src/script/lua_api/l_http.cpp

```diff
@@ -49,16 +49,44 @@ void ModApiHttp::read_http_fetch_request(lua_State *L, HTTPFetchRequest &req)
 	req.multipart = getboolfield_default(L, 1, "multipart", false);
 	req.timeout = getintfield_default(L, 1, "timeout", 3) * 1000;
 
-	// post_data: if table, post form data, otherwise raw data
+	// post_data: if table, post form data, otherwise raw data DEPRECATED
 	lua_getfield(L, 1, "post_data");
 	if (lua_istable(L, 2)) {
+		req.method = HTTP_POST;
 		lua_pushnil(L);
 		while (lua_next(L, 2) != 0) {
-			req.post_fields[readParam<std::string>(L, -2)] = readParam<std::string>(L, -1);
+			req.fields[readParam<std::string>(L, -2)] = readParam<std::string>(L, -1);
 			lua_pop(L, 1);
 		}
 	} else if (lua_isstring(L, 2)) {
-		req.post_data = readParam<std::string>(L, 2);
+		req.method = HTTP_POST;
+		req.raw_data = readParam<std::string>(L, 2);
+	}
+	lua_pop(L, 1);
+
+	lua_getfield(L, 1, "method");
+	if (lua_isstring(L, -1)) {
+		std::string mth = getstringfield_default(L, 1, "method", "");
+		if (mth == "GET")
+			req.method = HTTP_GET;
+		else if (mth == "POST")
+			req.method = HTTP_POST;
+		else if (mth == "PUT")
+			req.method = HTTP_PUT;
+		else if (mth == "DELETE")
+			req.method = HTTP_DELETE;
+	}
+	lua_pop(L, 1);
+
+	lua_getfield(L, 1, "data");
+	if (lua_istable(L, 2)) {
```

Comment:
`
local force_post = false;
lua_getfield(L, 1, "data");
if (lua_isnil(L, 2)) {
	lua_pop(L, 1);
	// Fallback to legacy
	lua_getfield(L, 1, "post_data");
	force_post = true;
}

Reply:
Good idea, did that!